src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,233 @@
+'use strict'
+var BN = require('ethereumjs-util').BN
+var utileth = require('ethereumjs-util')
+var varUtil = require('./variable')
+
+module.exports = {
+  decodeArray: function (type, storage, location) {
+    return this.decodeArrayItems(type, storage, 0, location)
+  },
+
+  decodeArrayItems: function (type, storage, depth, location) {
```

Comment:
I'm sure there is a common sections, such as encoding of uint. With ES6 classes, we could have a base class for arrays and three implementations (calldata, memory, storage) - if that makes sense code wise.